adapter/internal/oasparser/model/async_api.go

```diff
@@ -0,0 +1,160 @@
+)
+
+// AsyncAPI is the struct for the AsyncAPI 2.0.0 definition
+type AsyncAPI struct {
+	SpecVersion string `json:"asyncapi,omitempty"`
+	ID          string `json:"id,omitempty"`
+	Info        struct {
+		Title   string `json:"title,omitempty"`
+		Version string `json:"version,omitempty"`
+	} `json:"info,omitempty"`
+	Servers struct {
+		Production Server `json:"production,omitempty"`
+		Sandbox    Server `json:"sandbox,omitempty"`
+	} `json:"servers,omitempty"`
+	Channels   map[string]ChannelItem `json:"channels,omitempty"`
+	Components struct {
+		Schemas         map[string]interface{}    `json:"schemas,omitempty"`
+		Messages        map[string]interface{}    `json:"messages,omitempty"`
+		SecuritySchemes map[string]SecurityScheme `json:"securitySchemes,omitempty"`
+	} `json:"components,omitempty"`
+	VendorExtensions map[string]interface{} `json:"-"`
+}
+
+// Server object in AsyncAPI
+type Server struct {
+	URL             string                 `json:"url,omitempty"`
+	Protocol        string                 `json:"protocol,omitempty"`
+	ProtocolVersion string                 `json:"protocolVersion,omitempty"`
+	Variables       map[string]interface{} `json:"variables,omitempty"`
+	Security        []map[string][]string  `json:"security,omitempty"`
+	Bindings        map[string]interface{} `json:"bindings,omitempty"`
+}
+
+// ChannelItem in AsyncAPI channels
+type ChannelItem struct {
+	Ref              string                 `json:"$ref,omitempty"`
+	Subscribe        interface{}            `json:"subscribe,omitempty"` // TODO: (suksw) OperationAsync or $ref
+	Publish          interface{}            `json:"publish,omitempty"`   // TODO: (suksw) OperationAsync or $ref
+	Parameters       map[string]interface{} `json:"parameters,omitempty"`
+	Bindings         map[string]interface{} `json:"bindings,omitempty"`
+	VendorExtensions map[string]interface{} `json:"-"`
+}
+
+// OperationAsync is the Operation object that includes the message object
+type OperationAsync struct {
+	OperationID string `json:"operationId,omitempty"`
+	Message     struct {
+		Headers       string `json:"headers,omitempty"`
+		Payload       string `json:"payload,omitempty"`
+		CorrelationID string `json:"correlationId,omitempty"`
+		SchemaFormat  string `json:"schemaFormat,omitempty"`
+		ContentType   string `json:"contentType,omitempty"`
+		Name          string `json:"name,omitempty"`
+		Title         string `json:"title,omitempty"`
+	}
+}
+
+// SetInfoAsyncAPI populates the MgwSwagger object with information in asyncapi.yaml.
+func (swagger *MgwSwagger) SetInfoAsyncAPI(asyncAPI AsyncAPI) error {
+	swagger.vendorExtensions = asyncAPI.VendorExtensions
+	swagger.securityScheme = asyncAPI.getSecuritySchemes()
+	swagger.resources = asyncAPI.getResources()
+
+	if asyncAPI.Servers.Production.URL != "" {
+		endpoint, err := getWebSocketEndpoint(asyncAPI.Servers.Production.URL)
+		if err == nil {
+			productionEndpoints := append([]Endpoint{}, *endpoint)
```

Comment:
Don't we support endpoint security? we can support userpassword security type, aren't we?

Reply:
We are supposed to. Thanks for noticing!

Reply:
Found that they get set at the following locations. 
- https://github.com/wso2/product-microgateway/blob/0bbb23b52d72e07c61a0db036710fbf51a0a8774/adapter/internal/oasparser/model/mgw_swagger.go#L1226
- https://github.com/wso2/product-microgateway/blob/0bbb23b52d72e07c61a0db036710fbf51a0a8774/adapter/internal/discovery/xds/server.go#L317